tests/acceptance-tests/src/acceptance/http_forwarded_for_test.go

```diff
@@ -0,0 +1,57 @@
+package acceptance_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gexec"
+
+	"github.com/cloudfoundry-incubator/cf-test-helpers/cf"
+	"github.com/cloudfoundry-incubator/cf-test-helpers/generator"
+	"github.com/cloudfoundry-incubator/cf-test-helpers/helpers"
+
+	"encoding/json"
+	"net"
+	"strings"
+)
+
+type Headers struct {
+	X_Forwarded_For   []string `json:"X-Forwarded-For"`
+	X_Forwarded_Proto []string `json:"X-Forwarded-Proto"`
+}
+
+var _ = PDescribe("X-Forwarded headers", func() {
+	var headers Headers
+
+	var _ = BeforeEach(func() {
+		appName := generator.PrefixedRandomName("CATS-APP-")
+		Expect(cf.Cf(
+			"push", appName,
+			"-b", config.GoBuildpackName,
+			"-p", "../../example_apps/print_request_headers",
+			"-d", config.AppsDomain,
+			"-c", "./bin/debug_app; sleep 1; echo 'done'",
+		).Wait(CF_PUSH_TIMEOUT)).To(Exit(0))
+
+		jsonData := helpers.CurlAppRoot(appName)
+		headers = Headers{}
+
+		err := json.Unmarshal([]byte(jsonData), &headers)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("should have an upstream IP that is not on our network", func() {
+		Expect(headers.X_Forwarded_For).Should(HaveLen(1))
```

Comment:
@alext is correct - one of the odder quirks of HTTP is allowing multiple headers of the same type if the field is comma separated. Given that it's not well defined which fields _are_ comma separated, generally it's easier to just let the implementer decide. Yes, this means you nearly always get unexpected behaviour.

> Multiple message-header fields with the same field-name MAY be present in a message if and only if the entire field-value for that header field is defined as a comma-separated list [i.e., #(values)]. It MUST be possible to combine the multiple header fields into one "field-name: field-value" pair, without changing the semantics of the message, by appending each subsequent field-value to the first, each separated by a comma.